src/surveySimPP/readers/ObjectDataReader.py

```diff
@@ -1,19 +1,42 @@
-"""Classes for reading object-related data from a variety of sources.
-Each data source needs to have a column ObjID that identifies the object and
-can be used for joining and filtering.
+"""Base class for reading object-related data from a variety of sources
+and returning a pandas data frame.
+
+Each subclass of ObjectDataReader must implement at least the functions
+_read_rows_internal and _read_objects_internal, both of which return a
+pandas data frame. Each data source needs to have a column ObjID that
+identifies the object and can be used for joining and filtering.
+
+Caching is implemented in the base class. This will lazy load the full
+table into memory from the chosen data source, so it should only be
+used with smaller data sets. Both``read_rows`` and ``read_objects``
```

Comment:
minor: White space between Both and "``read_rows``.

Reply:
Fixed.